sockeye/decoder.py

```diff
@@ -249,62 +236,98 @@ def _decode(self,
 
         for layer in self.layers:
             target = layer(target=target,
-                           target_max_length=target_max_length,
                            target_bias=target_bias,
                            source=source_encoded,
-                           source_lengths=source_encoded_lengths,
-                           source_max_length=source_encoded_max_length)
+                           source_bias=source_bias)
         target = self.final_process(data=target, prev=None)
 
         return target
 
     def decode_step(self,
-                    target_embed: mx.sym.Symbol,
-                    target_embed_lengths: mx.sym.Symbol,
-                    target_embed_max_length: int,
+                    step: int,
                     target_embed_prev: mx.sym.Symbol,
                     source_encoded_max_length: int,
                     *states: mx.sym.Symbol) -> Tuple[mx.sym.Symbol, mx.sym.Symbol, List[mx.sym.Symbol]]:
         """
-        Decodes a single time step given the embedded target sequence and previous decoder states.
+        Decodes a single time step given the current step, the previous embedded target word,
+        and previous decoder states.
         Returns decoder representation for the next prediction, attention probabilities, and next decoder states.
         Implementations can maintain an arbitrary number of states.
 
-        :param target_embed: Embedded target sequence. Shape: (batch_size, target_embed_max_length, target_num_embed).
-        :param target_embed_lengths: Lengths of embedded target sequences. Shape: (batch_size,).
-        :param target_embed_max_length: Size of embedded target sequence dimension.
+        :param step: Global step of inference procedure.
         :param target_embed_prev: Previous target word embedding. Shape: (batch_size, target_num_embed).
         :param source_encoded_max_length: Length of encoded source time dimension.
         :param states: Arbitrary list of decoder states.
         :return: logit inputs, attention probabilities, next decoder states.
         """
-        source_encoded, source_encoded_lengths = states
+        source_encoded, source_encoded_lengths = states[:2]  # pylint: disable=unbalanced-tuple-unpacking
```

Comment:
Perhaps a comment reminding us what `states` should contain since we're confidently taking the first two elements?

Reply:
done